bin/date

```diff
@@ -80,8 +80,19 @@ sub run {
 	}
 
 sub select_format {
-	my( $supplied ) = grep { /\A\+/ } @_;
-	return $1 if $supplied =~ /\A\+(.+)/;
+	my $supplied;
```

Comment:
I ended up backing out this change. I should have paid attention to the test failing, but there was another test failing so I just ignored both. That's dumb.

A few things to note here:

* don't exit from utility subs. In the fix I die and wrap the sub in an eval. That gives the main code a chance to ignore it. Note a huge deal here because it's all private code, but in general it's about separation of concerns.
* we can't stop if there is an option that doesn't start with `+` because there are other options possible, and maybe we need better argument processing so nothing else is left.